lib/fluent/plugin/in_kafka_group.rb

```diff
@@ -175,14 +180,21 @@ def reconnect_consumer
     log.warn "Stopping Consumer"
     consumer = @consumer
     @consumer = nil
-    consumer.stop
-    log.warn "Could not connect to broker. Next retry will be in #{@retry_wait_seconds} seconds"
+    if consumer 
+      consumer.stop
+    end
+    log.warn "Could not connect to broker. retry_time:#{@retry_count}. Next retry will be in #{@retry_wait_seconds} seconds"
+    @retry_count = @retry_count + 1
     sleep @retry_wait_seconds
     @consumer = setup_consumer
     log.warn "Re-starting consumer #{Time.now.to_s}"
+    @retry_count = 0 
   rescue =>e
     log.error "unexpected error during re-starting consumer object access", :error => e.to_s
     log.error_backtrace
+    if @retry_count <= retry_limit or disable_retry_limit == true
```

Comment:
`== true` is not needed because `disable_retry_limit` is boolean.
In addition, `@retry_limit` is better than `retry_limit`.